### PR TITLE
Add admin/retention methods

### LINKED ISSUE #330: Implement Admin Retention API
Mastodon's Admin Retention API allows an admin to display retention data over time. The goal of this task is to implement the following endpoints of this API in BigBone:

- Calculate retention data

The Admin Retention API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/retention/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

This PR adds the methods under the admin/retention endpoint which currently only contains [“Calculate retention data”](https://docs.joinmastodon.org/methods/admin/retention/#create).

I had to change the date/time parser slightly. For some reason, for this API, the Mastodon API documentation contains dates formatted with an explicit offset (`2022-09-01T00:00:00+00:00`) instead of the UTC `Z` (`2022-09-01T00:00:00Z`) we previously saw in all other places.
For that reason, I have replaced `Instant.parse` which implicitly parses with the `DateTimeFormatter.ISO_INSTANT` with `DateTimeFormatter.ISO_OFFSET_DATE_TIME` which is able to handle both explicit offset (like `+00:00`) but also the `Z`.

Closes #330 

# Type of Change

(Keep the one that applies, remove the rest)

- Bug fix (non-breaking change which fixes an issue)
  - Not necessarily a bug we would have encountered yet, but we’re now also able to parse offset date time strings
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
  - [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 

# Mandatory Checklist

- [x] My change follows the projects coding style
- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] KDoc added to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- In case you worked on new features: Did you also implement the reactive endpoint (bigbone-rx)? ✅ 
- In case you added new *Methods classes: Did you also add it to `MastodonClient`? ✅ 
- In case you worked on endpoints: Please mention in the PR that [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated (if needed)

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+         */
+        @SerialName("date")
+        @Serializable(with = DateTimeSerializer::class)
+        val date: PrecisionDateTime = PrecisionDateTime.InvalidPrecisionDateTime.Unavailable,
+
+        /**
+         * The percentage rate of users who registered in the specified period and were active for the given date bucket.
+         */
+        @SerialName("rate")
+        val rate: Double? = null,
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:73]: A `Float` is probably enough here, isn't it?

File: bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        /**
+         * How many users registered in the specified period and were active for the given date bucket.
+         */
+        @SerialName("value")
+        val value: String? = null
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: Looking at the Mastodon documentation, I'd say this should be of type `Int`.

**Review by andregasser — CHANGES_REQUESTED** [review]
It is looking good, just added two minor comments 👍

File: bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt (hunk 2 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: From what it says in the documentation, I’d agree, but the examples all have Strings 😕

Comment (bocops) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: The [source for the cohort data](https://github.com/mastodon/mastodon/blob/1416745a2b23f95dcdcc459faa1367525d53e69d/app/lib/admin/metrics/retention.rb#L58) uses `.to_s` on the internal value, which implies that String is correct and the documentation is wrong (again).

Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: Can we create a PR to fix the documentation? 🤔

Comment (andregasser) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: Sure we can: https://github.com/mastodon/documentation :-)

Comment (andregasser) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminCohort.kt:79]: Let's keep this as a `String` then 👍

**Review by andregasser — APPROVED** [review]
👍

**Pull request merged**